### ISSUE TITLE
Unexpected Commanders Act pause / play transitions when playback starts?

### ISSUE THREAD

**Issue by defagos** [SRGSSR/srganalytics-apple#84]
@GoetzMar reported unexpected pause / play transitions at playback starts observed when playing [The Written Face](https://www.playsuisse.ch/watch/2083918?locale=fr) in Play Suisse.

<img width="574" alt="image001" src="https://github.com/SRGSSR/srganalytics-apple/assets/170201/e3841c03-4330-440c-89ca-4815c0781b14">

<img width="477" alt="image002" src="https://github.com/SRGSSR/srganalytics-apple/assets/170201/f86c220e-0ca3-43bb-a8c0-b26f0801cf08">

Events are not displayed in order here, please check the complete [log](https://github.com/SRGSSR/srganalytics-apple/files/14010014/log_ca_video_urn_rio_video_2083918.txt) for more information.

People conjecture this might be related to initial content buffering.

### Issue type

Incorrect behavior

### Environment information

As extracted from the logs:

* Library version: 9.0.0
* iOS / tvOS version: 16.3
* Device: iPhone 13

### Reproducibility

Difficult to reproduce

**Comment by defagos** [comment]
# Analysis

No major change has been made to SRG Analytics recently, but looking at the source code for version 9.0.0 we can say that this cannot related to buffering:

- The `SRGMediaPlayerPlaybackStateStalled` is simply [ignored](https://github.com/SRGSSR/srganalytics-apple/blob/9.0.0/Sources/SRGAnalyticsMediaPlayer/SRGMediaPlayerTracker.m#L132-L155). It can never lead to additional play / pause transitions.
- Events are emitted when precise [allowed state transitions](https://github.com/SRGSSR/srganalytics-apple/blob/9.0.0/Sources/SRGAnalyticsMediaPlayer/SRGMediaPlayerTracker.m#L176-L180) happen. With the log we have the additional events must be related to changes betweeen `SRGMediaPlayerPlaybackStatePlaying` and `SRGMediaPlayerPlaybackStatePaused`.
- At the SRG Media Player level such playback state changes occur [when the player rate is set to zero / a positive value](https://github.com/SRGSSR/srgmediaplayer-apple/blob/develop/Sources/SRGMediaPlayer/SRGMediaPlayerController.m#L265-L290). Rate changes might be the result of a call to a pause method or because the system, for some reason, alters the rate. Usually this should not happen during playback, except if the app is backgrounded or if another app steals the audio focus.

# Next steps

To understand where the additional pause comes from we have to understand why playback is briefly paused which, without additional context, cannot be reproduced. Having tested the affected Play Suisse stream we namely cannot see anything special and the code itself is clear.

We need a precise use case if we want to further investigate the call stack leading to the additional pause. In the meantime the reported additional pause / play transition does not produce incorrect results. Moreover ,if the issue is rare (and it is likely the case since this is the first time we hear about this issue), it should also not raise tracking costs too much, but we have not been provided with numbers which would let us evaluate how many times this happens in practice.

**Comment by GoetzMar** [comment]
Thanks for the feedback.
I agree that most likely it won't be to big of a case since this is the first time I found it and I already tested a lot of apps with media events. But this behavior happened on multiple videos when I tested it and on the one example that got tested by the Mapp consultant.

What would be the best way to provide you the information needed to reproduce it? Maybe a quick call on Friday helps, so I could show you the case live?
